server/src/main/java/org/elasticsearch/cluster/routing/RoutingNode.java

```diff
@@ -43,6 +43,8 @@ public class RoutingNode implements Iterable<ShardRouting> {
 
     private final LinkedHashSet<ShardRouting> relocatingShards;
 
+    private final LinkedHashSet<ShardRouting> startedShards;
```

Comment:
can we add some tests to check that the started shards are handled correctly in this class?

Reply:
Good idea, I've added a test for checking that the `started` method accounts for added and removed nodes.